### ISSUE TITLE
toEqual giving false positives when comparing objects in arrays

### ISSUE THREAD

**Issue by xbili** [mjackson/expect#53]
Hi there! I've been playing around with this for a bit and found out that `toEqual` returns false positives when comparing objects inside arrays.

Refer to the example below:
```js
let arr1 = [
  {
    id: 0,
    text: 'Array Object 0',
    boo: false
  },
  {
    id: 1,
    text: 'Array Object 1',
    boo: false
  }
];

let arr2 = [
  {
    id: 0,
    text: 'Array Object 0',
    boo: true // value of boo is changed to true here
  },
  {
    id: 1,
    text: 'Array Object 1',
    boo: false
  }
];

expect(arr1).toEqual(arr2);

console.log('All tests passed.');
```
The above test pass just fine, I am not sure if it's an intended behaviour. Anyone else facing similar issues? Thanks! :smile:

**Comment by Poplava** [comment]
I got the same problem when I tested ```arguments``` which contains objects.
I think it would be helpful to have ```deepEqual``` for arrays and array-like objects with complex data structures inside.

**Comment by mjackson** [comment]
I recently made the switch to [`is-equal`](https://github.com/ljharb/is-equal) from [`deep-equal`](https://github.com/substack/node-deep-equal) so that we can support ES2015 Map and Set comparisons, but it seems we have regressed in a few cases. I've added a failing test.

/cc @ljharb

**Comment by ljharb** [comment]
Sorry about this! I'll fix it ASAP.

**Comment by ljharb** [comment]
Fixed and released as `v1.4.2`.

**Comment by mjackson** [comment]
:tada: Thanks, @ljharb!

**Comment by mjackson** [comment]
Fixed and released in version 1.13.4

**Comment by xbili** [comment]
Thanks! @ljharb and @mjackson! :+1:
